gprofiler/python.py

```diff
@@ -245,18 +249,10 @@ def _dump(self) -> Path:
             return output
         except TimeoutError:
             # error flow :(
-            try:
-                if _reinitialize_profiler is not None:
-                    logger.warning("Reverting to py-spy")
-                    global _profiler_class
-                    _profiler_class = PySpyProfiler
-                    _reinitialize_profiler()
-            finally:
-                # in any case, we should stop PyPerf.
-                logger.warning("PyPerf dead/not responding, killing it")
-                process = self.process  # save it
-                self._terminate()
-                self._pyperf_error(process)
+            logger.warning("PyPerf dead/not responding, killing it")
+            process = self.process  # save it
+            self._terminate()
+            self._pyperf_error(process)
```

Comment:
Did you remove the try-finally for a reason ?

Reply:
Yes - the swap logic is now done in `PythonProfiler.snapshot`, so the `try` part actually happens later, and the `finally` part can happen immediately.